File: Algorithm/Leetcode/0013.cpp
// 1365. How Many Numbers Are Smaller Than the Current Number

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
		map<int, int> counter;
		for (auto n : nums) ++counter[n];
		
        int sum_tmp = {};
        map<int, int> sum_counter;
		for (auto& c : counter)
		{
            sum_counter[c.first] = sum_tmp;
			sum_tmp += c.second;			
		}

		for (size_t i = 0; i < nums.size(); i++)
		{
			nums[i] = sum_counter[nums[i]];
		}

		return nums;
        
    }
};

File: Algorithm/Leetcode/0000_0100/0011_0020/0013.py
# 1365. How Many Numbers Are Smaller Than the Current Number

"""https://leetcode.com/problems/how-many-numbers-are-smaller-than-the-current-number/discuss/524865/Clean-Python-3-sorting-and-counting"""

# 看不懂

class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        indices = {}
        for idx, num in enumerate(sorted(nums)):
            indices.setdefault(num, idx)
        return [indices[num] for num in nums]“”“